src/Microsoft.Azure.SignalR.Common/ClientInvocation/IClientResultsManager.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NET7_0_OR_GREATER
+using System;
+using Microsoft.Azure.SignalR.Protocol;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Buffers;
+using Microsoft.AspNetCore.SignalR.Protocol;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal interface IClientResultsManager
+    {
+        public ulong GetNewInvocation();
+
+        public void AddServiceMappingMessage(string invocationId, ServiceMappingMessage serviceMappingMessage);
+
+        public void RemoveServiceMappingMessageWithOfflinePing(string instanceId);
+
+        public Task<object> AddRoutedInvocation(string connectionId, string invocationId, string callerServerId, CancellationToken cancellationToken);
+
+        public Task<T> AddInvocation<T>(string connectionId, string invocationId, CancellationToken cancellationToken);
+
+        public void TryCompleteResult(string connectionId, string protocol, ReadOnlySequence<byte> message);
+
+        public void TryCompleteResult(string connectionId, CompletionMessage message);
+
+        public bool CheckRoutedInvocation(string invocationId);
+
+        public void TryCompleteRoutedResult(string connectionId, CompletionMessage message);
+
+        public (Type Type, string ConnectionId, object Tcs, Action<object, CompletionMessage> Completion)? RemoveInvocation(string invocationId);
```

Comment:
From an interface design perspective, the method naming needs to be readable and self explainable.

* How is serviceMappingMessage related to offline ping? Why AddServiceMappingMessage does not withOfflinePing?
* what is invocation and what is routedInvocation? why one has Add one does not? the difference is one has callServerId and one does not? 
* Is that the caller knows which is routed which is not? could it be the ClientResultManager to handle the logic? Will that make the logic simpler?

Reply:
- When a pod will go offline, all client invocations which were routed by this pod should be cleaned. These invocations are stored in their route server and original caller server. 
  - That's why `serviceMappingMessage` is related to offline ping. 
  - After a server receives an offline ping from a specific pod, this server should do 2 things
    -  For all invocations called by it, clean those were routed by the pod 
    -  For all invocations routed by it, clean those were routed by the pod
  - `AddServiceMappingMessage` doesn't involve in this procedure.
- The difference between invocation and routed invocation is whether the server is the original caller server for the invocation. 
- I designed a `ClientResultsManager`. It has 2 memebers `Caller` and `Router`. The `Caller` manages all client invocations which are originally created by the server. While the `Router` manages all invocations which are routed by the server.